app/scripts/contentscript.js

```diff
@@ -21,8 +23,19 @@ const inpageBundle = inpageContent + inpageSuffix
 // MetaMask will be much faster loading and performant on Firefox.
 
 if (shouldInjectProvider()) {
-  injectScript(inpageBundle)
-  start()
+  // If this is Brave, it requires coordination to know if we should be the
+  // web3 provider.
+  if (chrome.braveWallet && chrome.braveWallet.getWeb3Provider) {
```

Comment:
```suggestion
  if (chrome && chrome.braveWallet && chrome.braveWallet.getWeb3Provider) {
```
Correct me if I'm wrong, but I suspect that the `chrome` global wouldn't exist on Firefox? I think we need to check for that here too.

Reply:
Yeah, since `chrome` doesn't exist in firefox, this first clause would throw an undefined error. Either needs to be `typeof chrome !== 'undefined'` or you could import [extensionizer](https://www.npmjs.com/package/extensionizer) and check `extension.braveWallet` to avoid this problem.